stl/src/vector_algorithms.cpp

```diff
@@ -2356,52 +2397,19 @@ namespace {
 
-                const size_t _Needle_length = _Byte_length(_First2, _Last2);
-                const int _Needle_length_el = static_cast<int>(_Needle_length / sizeof(_Ty));
-
-                // Special handling of small needle
-                // The generic approach could also handle it but with worse performance
-                if (_Needle_length_el == 0) {
-                    return _Last1;
-                } else if (_Needle_length_el == 1) {
-                    _STL_UNREACHABLE; // This is expected to be forwarded to 'find' on an upper level
-                } else if (_Needle_length_el == 2) {
-                    return _Shuffle_impl<_Traits, 2>(_First1, _Last1, _First2, _Needle_length_el);
-                } else if (_Needle_length_el <= 4) {
-                    return _Shuffle_impl<_Traits, 4>(_First1, _Last1, _First2, _Needle_length_el);
-                } else if (_Needle_length_el <= 8) {
-                    if constexpr (sizeof(_Ty) == 4) {
-                        return _Shuffle_impl<_Traits, 8>(_First1, _Last1, _First2, _Needle_length_el);
-                    }
-                }
-
-                // Generic approach
-                const size_t _Needle_length_tail = _Needle_length & 0x1C;
-                const __m256i _Tail_mask         = _Avx2_tail_mask_32(_Needle_length_tail >> 2);
-
-                const void* _Stop2 = _First2;
-                _Advance_bytes(_Stop2, _Needle_length & ~size_t{0x1F});
+                const size_t _Needle_length         = _Byte_length(_First2, _Last2);
+                const size_t _Last_needle_length    = _Needle_length & 0x1F;
+                const size_t _Last_needle_length_el = _Last_needle_length / sizeof(_Ty);
 
-                for (auto _Ptr1 = static_cast<const _Ty*>(_First1); _Ptr1 != _Last1; ++_Ptr1) {
-                    const auto _Data1 = _Traits::_Set_avx(*_Ptr1);
-                    for (auto _Ptr2 = _First2; _Ptr2 != _Stop2; _Advance_bytes(_Ptr2, 32)) {
-                        const __m256i _Data2 = _mm256_loadu_si256(static_cast<const __m256i*>(_Ptr2));
-                        const __m256i _Eq    = _Traits::_Cmp_avx(_Data1, _Data2);
-                        if (!_mm256_testz_si256(_Eq, _Eq)) {
-                            return _Ptr1;
-                        }
-                    }
-
-                    if (_Needle_length_tail != 0) {
-                        const __m256i _Data2 = _mm256_maskload_epi32(static_cast<const int*>(_Stop2), _Tail_mask);
-                        const __m256i _Eq    = _Traits::_Cmp_avx(_Data1, _Data2);
-                        if (!_mm256_testz_si256(_Eq, _Tail_mask)) {
-                            return _Ptr1;
-                        }
-                    }
+                if (size_t _Needle_length_large = _Needle_length & ~size_t{0x1F}; _Needle_length_large != 0) {
+                    const void* _Stop2 = _First2;
+                    _Advance_bytes(_Stop2, _Needle_length & ~size_t{0x1F});

```

Comment:
Having extracted `_Needle_length_large`, we should use it instead of repeating `_Needle_length & ~size_t{0x1F}`.